aws/cloudformation.yml

```diff
@@ -0,0 +1,699 @@
+              command: !Sub |
+                #!/bin/bash -ex
+
+                echo "stage 1 update" >> /var/tmp/bootstrap.txt
+                apt-get update
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "stage 1 upgrade noninteractive" >> /var/tmp/bootstrap.txt
+                DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confnew" upgrade
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "signal trigger" >> /var/tmp/bootstrap.txt
+                /usr/local/bin/cfn-signal --success true --region ${AWS::Region} --stack ${AWS::StackName} --resource EC2InstancePrimary
+                echo "$?" >> /var/tmp/bootstrap.txt
+                #Warning: sudo cfn-signal seems to freeze/break the userdata.
+
+                echo "waiting for stack-create-complete" >> /var/tmp/bootstrap.txt
+                aws cloudformation wait stack-create-complete --stack-name ${AWS::StackName} --region ${AWS::Region}
+                echo "$?" >> /var/tmp/bootstrap.txt
+                echo "CFN Stack completed" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script1" >> /var/tmp/bootstrap.txt
+                bash -ex /var/tmp/stage1.sh  # Mastodon User/Docker/Docker-Compose
+                echo "script 1 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script2" >> /var/tmp/bootstrap.txt
+                sudo -u mastodon bash -ex /var/tmp/stage2.sh  # Mastodon
+                echo "script 2 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script3" >> /var/tmp/bootstrap.txt
+                bash -ex /var/tmp/stage3.sh  # Nginx and Certbot
+                echo "script 3 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script4" >> /var/tmp/bootstrap.txt
+                sudo -u mastodon bash -ex /var/tmp/stage4.sh  # Crontab/Maintenance Jobs
+                echo "script 2 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "Final reboot" >> /var/tmp/bootstrap.txt
+                systemctl reboot
+
+        create_files:
+          files: 
+            /var/tmp/stage1.sh:
+              content: !Sub |
+                #!/bin/bash -ex
+
+                echo "Create mastodon group" >> /var/tmp/bootstrap.txt
+                addgroup -gid 991 mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Create mastodon user" >> /var/tmp/bootstrap.txt
+                adduser --quiet --disabled-password --gid 991 --uid 991 mastodon --gecos GECOS
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Grant mastodon sudo" >> /var/tmp/bootstrap.txt
```

Comment:
Er, why?

Reply:
I'm not 100% sure what you mean? this is just following the production guide to match the GID/UID from the docker container.

Reply:
Sudoer is not required for the installation.. but I figured it be nice for the admin to have for maintenance.

Let's drop it for a little extra security posture :)